GabbyTam_Fatima_functionpractice.js

```diff
@@ -0,0 +1,238 @@
+https://www.freecodecamp.org/news/how-to-reverse-a-string-in-javascript-in-3-different-ways-75e4763c68cb/
+function reverseString(str) {
+    var newString = "";
+    for (var i = str.length - 1; i >= 0; i--) {
+        newString += str[i];
+        //console.log(newString);
+    }
+    if(newString== str){
+      return true;
+    } else {
+      return false;
+    }
+}
+
+// console.log(reverseString('mom'));
+reverseString('racecar');
+
+//Part 1: 1 Write a function calculateCube that takes a single number and prints the volume of a cube made from that number.
+
+function calculateCube(num) {
+  //console.log(num*num*num);
+  return Math.pow(num, 3);
+  //using built in exponent method 
+  //console.log(Math.pow(num,3));
+}
+console.log(calculateCube(2));
+
+
+// Part 1: 2 Write a function isAVowel that takes a character (i.e. a string of length 1) and returns true if it is a vowel, false otherwise. The vowel could be upper or lower case.
+// a, e, i, o, u
+function isAVowel(a) {
+  if (a === 'a' || a=== 'e' || a=== 'i'|| a=== 'o' || a=== 'u'){
+    return true;
+  } else {
+    return false;
+  }
+}
+
+//this works but its not efficient 
+console.log(isAVowel('e'));
+
+//Part 1: 3 Write a function getTwoLengths that accepts two parameters (strings). The function should return an array of numbers where each number is the length of the corresponding string.
+
+function getTwoLengths(first, second) {
+  let arrayNum= [];
+  arrayNum.push(first.length);
+  arrayNum.push(second.length);
+  return arrayNum;
+}
+//but why is in not .length-1?? 
+//answer: -1 and +1 comes into play when you are using for loops, declaring your variable at 0 or at 1 will make a difference as to what add on youll need to .length property 
+// .length counts string and arrays starting at index 0
+ 
+console.log(getTwoLengths("Hank", "Hippopopalous"));
+
+
+//Part 1: 4 Write a function getMultipleLengths that accepts a single parameter as an argument: an array of strings. The function should return an array of numbers where each number is the length of the corresponding string.
+
+const getMultipleLengths = (array) => {
+  let arrayOfNum= [];
+  for(let i=0; i<= array.length-1; i++){
+    arrayOfNum.push(array[i].length); 
+  }
+  return arrayOfNum;
+  //console.log(arrayOfNum);
+}
+
+console.log(getMultipleLengths(["hello", "what", "is", "up", "dude"]));
+
+//Part 2: 1 Define a function maxOfThree that takes three numbers as arguments and returns the largest of them. If all numbers are the same, it doesn't matter which one is returned. If the two largest numbers are the same, one of them should be returned.
+
+
+/*
+first attempt: 
+
+ const maxOfThree = (num1, num2, num3) => {
+  if(num1> num2 && num1> num3){
+    return num1;
+  }
+  if(num2> num1 && num2> num3){
+    return num2;
+  }
+  if(num3> num1 && num3 > num2){
+    return num3;
+  }
+  if(num1== num2== num3){
+    return num1;
+  }
+}
+
+console.log(maxOfThree(6, 9, 1));
+
+*/
```

Comment:
nice work here! a couple of things:
* in line 91, I think if you did `maxOfThree(5, 5, 5)`, it would have still given you undefined. That's because JS only reads boolean expressions that have 2 parts. If you want to compare num2 to num3, you'd have to do something like `if(num1 === num2 && num2 === num3)`. 
* this doesn't return a value for something like `maxOfThree(1, 5, 5)` or `maxOfThree(5, 1, 5)`. Try replacing a few of your `>` in your conditionals in lines 82, 85, and 88 with `>=`.